src/app.rs

```diff
@@ -451,6 +457,23 @@ where
     }
 }
 
+fn extract_auth_header(headers: HeaderMap) -> Option<BTreeMap<String, String>> {
```

Comment:
Can you add a comment for this function explaining why this is useful?

Reply:
Added